src/OpenTelemetry/OpenTelemetrySdk.cs

```diff
@@ -0,0 +1,124 @@
+
+using System.Diagnostics;
+using Microsoft.Extensions.DependencyInjection;
+using OpenTelemetry.Internal;
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Trace;
+
+namespace OpenTelemetry;
+
+/// <summary>
+/// Contains methods for configuring the OpenTelemetry SDK and accessing
+/// logging, metrics, and tracing providers.
+/// </summary>
+public sealed class OpenTelemetrySdk : IDisposable
+{
+    private readonly ServiceProvider serviceProvider;
+
+    private OpenTelemetrySdk(
+        Action<IOpenTelemetryBuilder> configure)
+    {
+        Debug.Assert(configure != null, "configure was null");
+
+        var services = new ServiceCollection();
+
+        var builder = new OpenTelemetrySdkBuilder(services);
+
+        configure!(builder);
+
+        this.serviceProvider = services.BuildServiceProvider();
+
+        this.LoggerProvider = (LoggerProvider?)this.serviceProvider.GetService(typeof(LoggerProvider))
+            ?? new NoopLoggerProvider();
+        this.MeterProvider = (MeterProvider?)this.serviceProvider.GetService(typeof(MeterProvider))
+            ?? new NoopMeterProvider();
+        this.TracerProvider = (TracerProvider?)this.serviceProvider.GetService(typeof(TracerProvider))
+            ?? new NoopTracerProvider();
+    }
+
+    /// <summary>
+    /// Gets the <see cref="Logs.LoggerProvider"/>.
+    /// </summary>
+    /// <remarks>
+    /// Note: The default <see cref="LoggerProvider"/> will be a no-op instance.
+    /// Call <see
+    /// cref="OpenTelemetryBuilderSdkExtensions.WithLogging(IOpenTelemetryBuilder)"/> to
+    /// enable logging.
+    /// </remarks>
+    public LoggerProvider LoggerProvider { get; }
+
+    /// <summary>
+    /// Gets the <see cref="Metrics.MeterProvider"/>.
+    /// </summary>
+    /// <remarks>
+    /// Note: The default <see cref="MeterProvider"/> will be a no-op instance.
+    /// Call <see
+    /// cref="OpenTelemetryBuilderSdkExtensions.WithMetrics(IOpenTelemetryBuilder)"/>
+    /// to enable metrics.
+    /// </remarks>
+    public MeterProvider MeterProvider { get; }
+
+    /// <summary>
+    /// Gets the <see cref="Trace.TracerProvider"/>.
+    /// </summary>
+    /// <remarks>
+    /// Note: The default <see cref="TracerProvider"/> will be a no-op instance.
+    /// Call <see
+    /// cref="OpenTelemetryBuilderSdkExtensions.WithTracing(IOpenTelemetryBuilder)"/>
+    /// to enable tracing.
+    /// </remarks>
+    public TracerProvider TracerProvider { get; }
+
+    /// <summary>
+    /// Gets the <see cref="IServiceProvider"/> containing SDK services.
+    /// </summary>
+    internal IServiceProvider Services => this.serviceProvider;
+
+    /// <summary>
+    /// Create an <see cref="OpenTelemetrySdk"/> instance.
+    /// </summary>
+    /// <param name="configure"><see cref="IOpenTelemetryBuilder"/> configuration delegate.</param>
+    /// <returns>Created <see cref="OpenTelemetrySdk"/>.</returns>
+    public static OpenTelemetrySdk Create(
```

Comment:
@rajkumar-rangaraj 

You commented: https://github.com/open-telemetry/opentelemetry-dotnet/pull/5325#issuecomment-2021600449

I'm dropping a review for the discussion so replies won't get sprinkled all over the PR.

> If we proceed with this approach, the simultaneous use of OpenTelemetrySdk.Create() and Sdk.CreateTracerProviderBuilder() should not be allowed. If both methods are used, one should throw an exception to clearly state that these APIs cannot be combined.

Doing that might be dangerous. Let's say someone creates a library. They want to send off telemetry from their library to their own servers. So they have a `TracerProvider` to do that. I would never recommend a library do this, but we have seen it 🤣 Now the host application comes along and uses `OpenTelemetrySdk.Create` to send its own telemetry somewhere. If we did this, suddenly the library would start to blow up. Or worse, let's say host is doing `Sdk.CreateTracerProviderBuilder()` and they upgrade some library which begins doing `OpenTelemetrySdk.Create` and suddenly blows up the host.

The initial concern over...

```csharp
        using var openTelemetry = OpenTelemetrySdk.Create(builder => builder
            .ConfigureResource(r => r.AddService("MyService"))
            .UseOtlpExporter());

        using var tracerProvider = Sdk.CreateTracerProviderBuilder()
            .AddAspNetCoreInstrumentation()
            .Build();
```

My question is, do we really have an issue here? Users can already find themselves in odd situations today:

```csharp
        using var tracerProviderA = Sdk.CreateTracerProviderBuilder()
            .AddOtlpExporter()
            .Build();

        using var tracerProviderB = Sdk.CreateTracerProviderBuilder()
            .AddAspNetCoreInstrumentation()
            .Build();
```

That is basically the same thing you are concerned about, no?

What we could do is `Obsolete` the "Sdk.Create*" APIs in favor of `OpenTelemetrySdk.Create`. I don't think it really solves the potential for users to shoot themselves in the foot, but it might help surface having the old API lingering around during migration to the new API?